lib/trino-parquet/src/main/java/io/trino/parquet/predicate/TupleDomainParquetPredicate.java

```diff
@@ -260,26 +306,88 @@ public static Domain getDomain(
+                List<Double> maximums = converter.getValuesAsDouble(type, columnIndex.getMaxValues(), columnName);
+                if (!maximums.isEmpty()) {
+                    return createDomain(type, columnIndex, id, columnName, hasNullValue, minimums, maximums, (BiFunction<Double, Double, ParquetDoubleStatistics>) (min, max) -> new ParquetDoubleStatistics(min, max));
+                }
+            }
+        }
+        else if (descriptor.getType().equals(PrimitiveTypeName.BINARY)
+                || (descriptor.getType().equals(PrimitiveTypeName.FIXED_LEN_BYTE_ARRAY)
+                    && type instanceof DecimalType
+                    && !((DecimalType) type).isShort())) {
+            List<Slice> minimums = converter.getValuesAsSlice(type, columnIndex.getMinValues());
+            if (!minimums.isEmpty()) {
+                List<Slice> maximums = converter.getValuesAsSlice(type, columnIndex.getMaxValues());
+                if (!maximums.isEmpty()) {
+                    return createDomain(type, columnIndex, id, columnName, hasNullValue, minimums, maximums, (BiFunction<Slice, Slice, ParquetSliceStatistics>) (min, max) -> new ParquetSliceStatistics(min, max));
+                }
             }
-            return Optional.of(new ParquetIntegerStatistics(longStatistics.genericGetMin(), longStatistics.genericGetMax()));
         }
 
-        if (statistics instanceof IntStatistics) {
+        return Domain.create(ValueSet.all(type), hasNullValue);
+    }
+
+    private static ParquetLongStatistics toParquetLongStatistics(Statistics<?> statistics, ParquetDataSourceId id, String column)
+    {
+        if (statistics instanceof LongStatistics) {
+            LongStatistics longStatistics = (LongStatistics) statistics;
+            return new ParquetLongStatistics(longStatistics.genericGetMin(), longStatistics.genericGetMax());
+        }
+        else if (statistics instanceof IntStatistics) {
             IntStatistics intStatistics = (IntStatistics) statistics;
-            if (intStatistics.genericGetMin() > intStatistics.genericGetMax()) {
-                throw corruptionException(column, id, intStatistics);
-            }
-            return Optional.of(new ParquetIntegerStatistics((long) intStatistics.getMin(), (long) intStatistics.getMax()));
+            return new ParquetLongStatistics((long) intStatistics.getMin(), (long) intStatistics.getMax());
+        }
+        else if (statistics instanceof BinaryStatistics) {
+            BinaryStatistics binaryStatistics = (BinaryStatistics) statistics;
+            Slice minSlice = Slices.wrappedBuffer(binaryStatistics.genericGetMin().getBytes());
+            Slice maxSlice = Slices.wrappedBuffer(binaryStatistics.genericGetMax().getBytes());
+            long minValue = getShortDecimalValue(minSlice.getBytes());
+            long maxValue = getShortDecimalValue(maxSlice.getBytes());
+            return new ParquetLongStatistics(minValue, maxValue);
```

Comment:
How do we know it's going to be a short decimal instead of a long decimal?

Reply:
The caller is required to check the type.